exporter/datadogexporter/internal/hostmetadata/provider/provider.go

```diff
@@ -22,15 +22,44 @@ type chainProvider struct {
 }
 
 func (p *chainProvider) Source(ctx context.Context) (source.Source, error) {
-	for _, source := range p.priorityList {
-		zapProvider := zap.String("provider", source)
+	// Auxiliary type for storing source provider replies
+	type reply struct {
+		src source.Source
+		err error
+	}
+
+	// Cancel all providers when exiting
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Run all providers in parallel
+	replies := make([]chan reply, len(p.priorityList))
+	for i, source := range p.priorityList {
 		provider := p.providers[source]
-		src, err := provider.Source(ctx)
-		if err == nil {
-			p.logger.Info("Resolved source", zapProvider, zap.Any("source", src))
-			return src, nil
+		replies[i] = make(chan reply)
+
+		go func(i int, source string) {
+			zapProvider := zap.String("provider", source)
+			p.logger.Debug("Trying out source provider", zapProvider)
+
+			src, err := provider.Source(ctx)
+			if err != nil {
+				p.logger.Debug("Unavailable source provider", zapProvider, zap.Error(err))
+			}
+
+			replies[i] <- reply{src: src, err: err}
+		}(i, source)
+	}
+
+	// Check provider responses in order to ensure priority
+	for i, ch := range replies {
+		reply := <-ch
+		if reply.err == nil {
```

Comment:
```suggestion
		if reply.err != nil {
		    continue
		}
```
Try to keep the indentation to the left

Reply:
Done!